ocean-client/src/programs/vault-maxi-program.ts

```diff
@@ -439,52 +447,75 @@ export class VaultMaxiProgram extends CommonProgram {
+          let shouldSend = false
+          if (this.isSingleMintA) {
+            let oracleA = this.getUsedOraclePrice(ALoan, false)
+            let oracleB = this.getUsedOraclePrice(
+              vault.collateralAmounts.find((coll) => coll.symbol === this.assetB),
+              true,
             )
-            const neededDusd = neededStock.multipliedBy(pool!.priceRatio.ba)
-            const stock_per_token = new BigNumber(pool!.tokenA.reserve).div(pool!.totalLiquidity.token)
-            const neededLPtokens = neededStock.div(stock_per_token)
-            if (
-              neededLPtokens.gt(lpTokens.amount) ||
-              neededDusd.gt(dusdLoan!.amount) ||
-              neededStock.gt(tokenLoan.amount)
-            ) {
+
+            const neededLPtokens = neededrepay
+              .times(safeRatio)
+              .times(pool.totalLiquidity.token)
+              .div(
+                BigNumber.sum(oracleA.times(pool.tokenA.reserve).times(safeRatio), oracleB.times(pool.tokenB.reserve)),
+              )
+
+            const neededAssetA = neededLPtokens.times(pool.tokenA.reserve).div(pool.totalLiquidity.token)
+            if (neededLPtokens.gt(lpTokens.amount) || neededAssetA.gt(ALoan!.amount)) {
               message +=
                 `would need ${neededLPtokens.toFixed(4)} but got ${(+lpTokens.amount).toFixed(4)} ${
                   lpTokens.symbol
                 }.\n` +
-                `would need ${neededDusd.toFixed(1)}  but got ${(+dusdLoan!.amount).toFixed(1)}  ${
-                  dusdLoan!.symbol
-                }.\n` +
-                `would need ${neededStock.toFixed(4)} but got ${(+tokenLoan.amount).toFixed(4)} ${tokenLoan.symbol}.\n`
-              await telegram.send(message, LogLevel.WARNING) //@krysh is this warning or error?
+                `would need ${neededAssetA.toFixed(4)} but got ${(+ALoan!.amount).toFixed(4)} ${ALoan!.symbol}.\n`
+              shouldSend = true
             }
-          } else {
-            let oracleA = this.getUsedOraclePrice(tokenLoan, false)
-            let oracleB = this.getUsedOraclePrice(
-              vault.collateralAmounts.find((coll) => coll.symbol === this.assetB),
+          } else if (this.isSingleMintB) {
+            // case stable-DUSD
+            let oracleB = new BigNumber(1)
+            let tokenA = this.getCollateralTokenByKey(this.assetA)!
+            let oracleA = this.getUsedOraclePrice(
+              { symbol: tokenA.token.symbolKey, id: tokenA.tokenId, activePrice: tokenA.activePrice },
               true,
             )
 
             const neededLPtokens = neededrepay
               .times(safeRatio)
               .times(pool.totalLiquidity.token)
               .div(
-                BigNumber.sum(oracleA.times(pool.tokenA.reserve).times(safeRatio), oracleB.times(pool.tokenB.reserve)),
+                BigNumber.sum(oracleA.times(pool.tokenA.reserve), oracleB.times(pool.tokenB.reserve).times(safeRatio)),
```

Comment:
I guess it should be fine, but just to be sure - that this change is intended.

Reply:
ah nvm, this is the new handling for stable-coin pools - isn't it?

Reply:
yeah, the whole "else if(singleMintB)" is new. this line actually didn't change, but is new. The "original" line which is still the same is 462, which is marked by github as new.